ejb/WFLY-17986.adoc

```diff
@@ -0,0 +1,234 @@
+= Support external EJB client access to EAP deployments on OpenShift over http
+:author:            Richard Achmatowicz
+:email:             rachmato@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+Existing EAP customers use the EJB client library to access EJB-based deployments (SFSBs, SLSBs, MDBs) on EAP
+on bare metal and such access needs to be provided whether the EAP deployment is on bare metal or on OpenShift.
+
+This analysis document aims to highlight the issues and concerns in providing EJB client access to EJB-based
+deployments on EAP in OpenShift.
+
+=== Client location and protocol
+Generally, an EJB client application may be internal or external to the OpenShift cluster, and may use any one of
+three transport protocols supported by the EJB client: remote, remote+http, or http (or their encrypted variants).
+
+This issue specifically focuses on the case where the EJB client application is:
+
+* external to the OpenShift cluster
+* the transport protocol is http (or its encrypted variant https)
+
+For the other location / protocol combinations, please see the related issues
+(https://issues.redhat.com/browse/EAP7-2063[EAP7-2063],  https://issues.redhat.com/browse/EAP7-2064[EAP7-2064],
+https://issues.redhat.com/browse/EAP7-2065[EAP7-2065])
+
+=== Access to cluster
+One issue of relevance here is how an external EJB client application gains entry to the OpenShift cluster.
+
+NOTE: We sometimes use the word `cluster` to refer to the OpenShift cluster in which the EAP application is
+deployed; at other times, to refer to the set of Pods which make up the EAP deployment when the
+HA (high availability) server profile is used. We will try to refer either to the OpenShift cluster or
+the EAP cluster to avoid ambiguity.
+
+OpenShift provides a limited set of entry points for external applications to gain access to the cluster. For binary
+protocols, NodePorts are used to gain access. For protocols based on HTTP, routes or ingress controllers
+are available. A route is a per-deployment load balancer which is accessible external to the cluster.
+An ingress controller is a load balancer external to the cluster which can be shared by multiple deployments.
+
+NOTE: External access points for a deployment may be set up automatically (as in the case of a `route`,
+when the application is deployed using the Wildfly operator) or may need to be set up manually.
```

Comment:
s/Wildfly/WildFly

This happens in several parts of the document.

Reply:
Fixed, thanks, Yeray!